perception/sandbox/eyehandcal/scripts/collect_data_and_cal.py

```diff
@@ -71,26 +71,23 @@ def robot_poses(ip_address, pose_generator, time_to_go=3):
     # Get reference state
     robot.go_home()
     for i, (pos_sampled, ori_sampled) in enumerate(pose_generator):
+        print( f"Moving to pose ({i}): pos={pos_sampled}, quat={ori_sampled.as_quat()}")
+
+        state_log = robot.move_to_ee_pose(position=pos_sampled,orientation=ori_sampled.as_quat(),time_to_go = time_to_go)
+        print(f"Length of state_log: {len(state_log)}")
+        if len(state_log) != time_to_go * robot.hz:
+            print(f"warning: log incorrect length. {len(state_log)} != {time_to_go * robot.hz}")
         while True:
-            print( f"Moving to pose ({i}): pos={pos_sampled}, quat={ori_sampled.as_quat()}")
-
-            state_log = robot.move_to_ee_pose(position=pos_sampled,orientation=ori_sampled.as_quat(),time_to_go = time_to_go)
-            print(f"Length of state_log: {len(state_log)}")
-            if len(state_log) != time_to_go * robot.hz:
```

Comment:
I did find the retry useful in case of (1) comm constraint errors, (2) not enough time to reach the goal ee position. When does it result in infinite loop?

Reply:
In my case, the `len(state_log)` is always 3 step shorter than `(time_to_go * robot.hz)`

Reply:
That seems like a Polymetis bug :\ We have tests which verify the episode length is correct. Perhaps file it as an issue and tag me/Austin